src/forklift/plugin.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+# * coding: utf8 *
+'''
+plugin.py
+
+A module the contains the plugin class that should be inherited from when building plugins
+'''
+
+import logging
+import settings
+from os.path import join
+
+
+class ScheduledUpdateBase(object):
+
+    def __init__(self):
+        #: the logging module to keep track of the plugin
+        self.log = logging.getLogger(settings.LOGGER)
+        #: the duration in hours that marks a plugin as dirty
+        self.expires_in_hours = 24
+        #: the path to where the data in dependencies resides
+        self.source_location = 'C:\\MapData\\SGID10.sde'
+        #: the parent path to where the output will be created
+        self.output_directory = 'C:\\MapData\\'
+        #: the file geodatabase where data will be inserted
+        self.gdb_name = 'SGID10.gdb'
+        #: the table names for all dependent data for an application
+        self.dependencies = []
+
+    def nightly(self):
+        '''This method will be called by forklift if the `expires_in_hours` value has expired.
+        This is the only method that needs to be implemented and will otherwise throw a `NotImplementedError`.
+        '''
+
+        raise NotImplementedError('Implement nightly in your child plugin.')
```

Comment:
So if the only thing that this plugin needs to do is a one-to-one update of feature classes from SDE, what would the code in here look like? Would it call core.py or something? Since this is such a common use case I wonder if we should include that logic in the base class?